components/TextFields/MDCTextInputBehavior.h

```diff
@@ -0,0 +1,104 @@
+/*
+ Copyright 2016-present the Material Components for iOS authors. All Rights
+ Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+@class MDCTextInput;
+
+/**
+ Material Design themed states for textInputs. The logic for 'automagic' error states changes:
+ underline color, underline text color.
+ https://www.google.com/design/spec/components/text-fields.html#text-fields-single-line-text-field
+ */
+@interface MDCTextInputBehavior : NSObject
+
+@property(nonatomic, nonnull, strong) id<MDCTextInput> textInput NS_SWIFT_NAME(input);
+
+/**
+ The text displayed in the underline text that explains the nature of the error.
+
+ The value of this property controls the state of the text input. When errorText != nil, the text
+ input is in an error state:
+ - The error text appears in a label below the input rectangle's underline with the errorColor as
+ text color.
+ - The input rectangle's underline is colored with the errorColor.
+
+ Set this property via setErrorText:errorAccessibilityValue:.
+ */
+@property(nonatomic, nullable, copy, readonly) NSString *errorText;
```

Comment:
why do we need to read the errorText? shouldn't the client app be able to read the underlineText?